eng/pipelines/common/global-build-job.yml

```diff
@@ -25,19 +25,12 @@ jobs:
     variables: ${{ parameters.variables }}
 
     steps:
-    - ${{ if eq(parameters.osGroup, 'OSX') }}:
-      - script: |
-          $(setScriptToEchoAndFailOnNonZero) 
-          brew install pkgconfig icu4c openssl autoconf automake libtool pkg-config python3
-          brew link --force icu4c
-          ln -s /usr/local/opt/openssl/lib/pkgconfig/libcrypto.pc /usr/local/lib/pkgconfig/
-          ln -s /usr/local/opt/openssl/lib/pkgconfig/libssl.pc /usr/local/lib/pkgconfig/
-          ln -s /usr/local/opt/openssl/lib/pkgconfig/openssl.pc /usr/local/lib/pkgconfig/
-        displayName: Install Build Dependencies
-
     - template: /eng/pipelines/common/clone-checkout-bundle-step.yml
 
-    - ${{ if in(parameters.osGroup, 'OSX', 'iOS','tvOS') }}: 
+    - ${{ if in(parameters.osGroup, 'OSX', 'iOS', 'tvOS') }}:
+      - script: sh $(Build.SourcesDirectory)/eng/install-native-dependencies.sh ${{ parameters.osGroup }}
```

Comment:
I think you accidentally brought back, `sh` into these steps.

Reply:
It was not accidental, see 7037603edce1a0d008685d209a9422beeacdf4ce which reverted the change. All the build were failing in Azure pipelines due to the removal of calling through `sh`.

Reply:
We have never needed the usage of sh, could you point me to the build error?

Reply:
I can't find the corresponding Azure build but here's the error I observed:
> /Users/runner/runners/2.166.4/work/_temp/56691a53-a761-472c-b9ab-a9b4eb004f5d.sh: line 1: /Users/runner/runners/2.166.4/work/1/s/eng/install-native-dependencies.sh: Permission denied

All the jobs running the  `install-native-dependencies.sh` script without running through `sh` would fail with this error. This is why I reverted it in 7037603edce1a0d008685d209a9422beeacdf4ce.

Note that I wrote the scripts to run through `sh` because all other `install-native-dependencies.sh` were run through `sh`, you can observe them on the master branch.

Reply:
```
wfurt-runtime furt$ ls -al eng/install-native-dependencies.sh
-rw-r--r--  1 furt  staff  1293 Apr 28 18:03 eng/install-native-dependencies.sh
```

it seems like that scrips does not have executable bit as it should @safern 
Perhaps something we should fix independently. (or lump here)

Reply:
Would you like to fix this in a follow up PR?

Reply:
@safern I'll leave it up to you.

Reply:
Ok, it would be nice to fix it (just to be consistent across all places where we invoke bash scripts in the yml), so if you have time that would be great.